src/services/recim/activity.ts

```diff
@@ -0,0 +1,88 @@
+import http from '../http';
+import { sport } from './sport';
+import { team } from './team';
+import { series } from './series';
+
+export type activity = {
+  ID: number;
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  RegistrationOpen: boolean;
+  SportID: sport;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+  Series: series[];
+  Team: team[];
+};
+
+type createdActivity = {
+  ID: number;
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  RegistrationOpen: boolean;
+  SportID: sport;
+  StatusID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Completed: boolean;
+};
+
+type uploadActivity = {
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  SportID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+};
+
+type patchActivity = {
+  Name: string;
+  RegistrationStart: string;
+  RegistrationEnd: string;
+  SportID: number;
+  StatusID: number;
+  MinCapacity: number;
+  MaxCapacity: number;
+  SoloRegistration: boolean;
+  Logo: string;
+  Completed: boolean;
+};
+
+//Activity Routes
+const createNewActivity = async (newActivity: uploadActivity): Promise<Object[]> => {
+  return await http.post('recim/activities', newActivity);
+};
```

Comment:
You can simplify this syntax significantly if you so choose:
```suggestion
const createNewActivity = (newActivity: uploadActivity): Promise<Object[]> =>  http.post('recim/activities', newActivity);
```

This is functionally equivalent because `http.post` returns a promise, so declaring async and `await`ing it (thereby wrapping the call to `http.post` in a Promise) is the same as just returning the interior Promise.

Also, an arrow function that is just one return statement can use the expression syntax `(params) => result` instead of the traditional statement syntax `(params) => { return result; }`.

The first is potentially a minor performance improvement, but ultimately neither make a huge difference and might potentially confuse developers who are less familiar with these concepts, so this condensed form is not **necessarily** better - although I personally do find it more readable.

So this is just a notice to make sure you're aware that this is an option in JS/TS, not a "you should write it this way" kind of suggestion.

Reply:
i see, I can make those changes.